miden-lib/src/tests/test_account.rs

```diff
@@ -326,13 +326,100 @@ fn test_set_item() {
     let _process = run_tx(&transaction).unwrap();
 }
 
-// TODO: reenable once storage map support is implemented
-#[ignore]
+// Test different account storage types
+#[test]
+fn test_get_storage_data_type() {
+    for storage_item in [storage_item_0(), storage_item_1(), storage_item_2()] {
+        let (tx_inputs, tx_args) =
+            mock_inputs(MockAccountType::StandardExisting, AssetPreservationStatus::Preserved);
+
+        let code = format!(
+            "
+        use.miden::kernels::tx::account
+        use.miden::kernels::tx::prologue
+
+        begin
+            # prepare the transaction
+            exec.prologue::prepare_transaction
+
+            # push the account storage item index
+            push.{item_index}
+
+            # get the data type of the respective storage slot
+            exec.account::get_storage_slot_type_info
+
+        end
+        ",
+            item_index = storage_item.index,
+        );
+
+        let transaction =
+            prepare_transaction(tx_inputs.clone(), tx_args.clone(), code.as_str(), None);
+        let _process = run_tx(&transaction).unwrap();
+        let transaction = prepare_transaction(tx_inputs, tx_args, &code, None);
+        let process = run_tx(&transaction).unwrap();
+
+        let storage_slot_data_type = match storage_item.slot.slot_type {
+            StorageSlotType::Value { value_arity } => (value_arity, 0),
+            StorageSlotType::Map { value_arity } => (value_arity, 1),
+            StorageSlotType::Array { value_arity, depth } => (value_arity, depth),
+        };
+
+        assert_eq!(process.get_stack_item(0), Felt::from(storage_slot_data_type.0));
+        assert_eq!(process.get_stack_item(1), Felt::from(storage_slot_data_type.1));
+    }
+}
+
 #[test]
 fn test_get_map_item() {
     let (tx_inputs, tx_args) =
         mock_inputs(MockAccountType::StandardExisting, AssetPreservationStatus::Preserved);
 
+    let storage_item = storage_item_2();
+    for (key, value) in STORAGE_LEAVES_2 {
+        let code = format!(
+            "
+            use.miden::account
+            use.miden::kernels::tx::prologue
+
+            begin
+                # prepare the transaction
+                exec.prologue::prepare_transaction
+
+                # push the item's KEY
+                push.{map_key}
+
+                # push the account storage item index
+                push.{item_index}
+
+                # get the map item
+                exec.account::get_map_item
+                
+            end
+            ",
+            item_index = storage_item.index,
+            map_key = prepare_word(&key),
+        );
+
+        let transaction =
+            prepare_transaction(tx_inputs.clone(), tx_args.clone(), code.as_str(), None);
+        let process = run_tx(&transaction).unwrap();
+        assert_eq!(value, process.get_stack_word(0));
```

Comment:
same changes for the stack assertion here.

Reply:
done